### PR TITLE
WIP: pre-pull-image-curl

### PR DESCRIPTION
**Pull request by MichalMaler**
Signed-off-by: Michal Maléř <mmaler@redhat.com>

> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

Adding a note and the CURL command to get the updated list of images for OpenShift pre-pull.

### CODE REVIEW

File: antora-playbook.yml [hunk 1]
```diff
@@ -96,12 +96,13 @@ asciidoc:
     prod-operator-image-name: che-operator
     prod-prev-ver-major: 6
     prod-prev-ver: 7.18
+    che-ver: 7.20.2
```
Comment (nickboldt) [inline antora-playbook.yml]: this version does not exist. See options at https://github.com/eclipse/che-operator/tree/master/olm/eclipse-che-preview-openshift/deploy/olm-catalog/eclipse-che-preview-openshift/

File: antora-playbook.yml [hunk 2]
```diff
@@ -96,12 +96,13 @@ asciidoc:
...
     prod-short: Che
     prod-upstream: Eclipse{nbsp}Che
     prod-url-secure: \https://che-host
     prod-url: \https://che-host:che-port
     prod-ver-major: 7
-    prod-ver: 7.19
+    prod-ver: 7.20
```
Comment (nickboldt) [inline antora-playbook.yml]: Latest release is 7.21, and latest che-ver is https://github.com/eclipse/che-operator/tree/master/olm/eclipse-che-preview-openshift/deploy/olm-catalog/eclipse-che-preview-openshift/7.21.1

File: modules/administration-guide/examples/che-prepulled-images-table.adoc [hunk 3]
```diff
@@ -22,4 +22,26 @@
...
+To get the latest build:
+
+. Use the following scriptlet:
++
+[subs="+attributes,+quotes"]
+----
+$➔ curl -sSLo- https://raw.githubusercontent.com/eclipse/che-operator/master/olm/eclipse-che-preview-openshift/deploy/olm-catalog/eclipse-che-preview-openshift/{che-ver}/eclipse-che-preview-openshift.v{che-ver}.clusterserviceversion.yaml | yq -r '.spec.relatedImages[].image' | sort | uniq
+----
++
+. Exclude redundant, unused digests, and unsupported images for OpenShift pre-pulling such as:
```
Comment (nickboldt) [inline modules/administration-guide/examples/che-prepulled-images-table.adoc]: This instruction is not going to be something a user can obviously/easily do.

To make it easier, perhaps we should instead use:

```
$➔ curl -sSLo- https://raw.githubusercontent.com/eclipse/che-operator/master/olm/eclipse-che-preview-openshift/deploy/olm-catalog/eclipse-che-preview-openshift/{che-ver}/eclipse-che-preview-openshift.v{che-ver}.clusterserviceversion.yaml | \
  yq -r '.spec.relatedImages[]
```

And then provide a snippet like:
```
...
{
  "name": "che-theia-7.19.2",
  "image": "quay.io/eclipse/che-theia@sha256:11b5fe33edd72df22f76d7933ba79029051f4c0c6adc6c1a91f2dfe4228455c9"
}
{
  "name": "che-theia-7.20.0",
  "image": "quay.io/eclipse/che-theia@sha256:92a8bac94e446f1f933d1a92bc96dace9450c9f4efdef5539a429a37d14be904"
}
{
  "name": "che-theia-7.20.1",
  "image": "quay.io/eclipse/che-theia@sha256:a56428ee28ceefca2a1cba51093f6015e0eee82659fec786f2e1fdc939fdb856"
}
...
```

With the instruction:

```
. Where multiple versions of the same image are listed, only the latest one needs to be pre-pulled. 

Note also that images like machine-exec, images without the `sleep` command and images that mount volumes cannot be pre-pulled, and thus need to be pulled by hand with `skopeo copy --all ...`.
```

File: antora-playbook.yml (hunk 1 above)
Comment (MichalMaler) [reply antora-playbook.yml]: Thx, for the link, updating .)

File: antora-playbook.yml (hunk 2 above)
Comment (MichalMaler) [reply antora-playbook.yml]: Updating

File: modules/administration-guide/examples/che-prepulled-images-table.adoc [hunk 4]
```diff
@@ -22,4 +22,26 @@
...
++
+* machine-exec image
+* images without a sleep command installed
+* images that mount volumes in the docker file
++
+For additional information, see link:https://github.com/eclipse/che/issues/17562[GitHub Issue 17562]
```
Comment (nickboldt) [inline modules/administration-guide/examples/che-prepulled-images-table.adoc]: Note that old versions of the same image will be removed soon. See https://github.com/eclipse/che-plugin-registry/pull/672 and https://github.com/eclipse/che/issues/18183

If you wait to merge this PR until after Che 7.22.0 is live, the instructions will be simpler.

Comment (MichalMaler) [reply modules/administration-guide/examples/che-prepulled-images-table.adoc]: OK, will prepare your suggestion and wait for the 7.22.0

**Comment by MichalMaler** [comment]
Merge this with 7.22 release ->  + update attributes to match 7.22 and most updated versions using
https://github.com/eclipse/che-operator/tree/master/olm/eclipse-che-preview-openshift/deploy/olm-catalog/eclipse-che-preview-openshift/

File: modules/administration-guide/examples/che-prepulled-images-table.adoc [hunk 5]
```diff
@@ -22,4 +22,53 @@
...
\ No newline at end of file
+|===
+
+[NOTE]
+====
+The table of recommended images to pre-pull is updated once per release and might vary from the most updated build at the time.
+
+To get the latest build:
+
+. Install the dependencies:
```
Comment (rkratky) [inline modules/administration-guide/examples/che-prepulled-images-table.adoc:33]: These should be prerequisites.

File: modules/administration-guide/examples/che-prepulled-images-table.adoc [hunk 6]
```diff
@@ -22,4 +22,53 @@
 |che-nodejs10-ubi |quay.io/eclipse/che-nodejs10-ubi:{prod-ver} |{prod-ver}
 
 |che-quarkus |quay.io/eclipse/che-quarkus:{prod-ver} |{prod-ver}
-|===
\ No newline at end of file
+|===
+
+[NOTE]
+====
+The table of recommended images to pre-pull is updated once per release and might vary from the most updated build at the time.
```
Comment (rkratky) [inline modules/administration-guide/examples/che-prepulled-images-table.adoc:29]: Let's get rid of the table completely and only offer the procedure below.

File: modules/administration-guide/examples/che-prepulled-images-table.adoc [hunk 7]
```diff
@@ -22,4 +22,53 @@
...
+
+. Use the following scriptlet:
++
+[subs="+attributes,+quotes"]
+----
+$ curl -sSLo- https://raw.githubusercontent.com/eclipse/che-operator/master/olm/eclipse-che-preview-openshift/deploy/olm-catalog/eclipse-che-preview-openshift/{che-ver}/eclipse-che-preview-openshift.v{che-ver}.clusterserviceversion.yaml | \
+  yq -r '.spec.relatedImages[]
+----
++
+----
```
Comment (rkratky) [inline modules/administration-guide/examples/che-prepulled-images-table.adoc:54]:
```suggestion
[source,json]
----
```

File: modules/administration-guide/examples/che-prepulled-images-table.adoc [hunk 8]
```diff
@@ -22,4 +22,53 @@
...
+...
```
Comment (rkratky) [inline modules/administration-guide/examples/che-prepulled-images-table.adoc]:
```suggestion
#[...]
```

File: modules/administration-guide/examples/che-prepulled-images-table.adoc [hunk 9]
```diff
@@ -22,4 +22,53 @@
...
+}
+{
+  "name": "che-theia-7.20.0",
+  "image": "quay.io/eclipse/che-theia@sha256:92a8bac94e446f1f933d1a92bc96dace9450c9f4efdef5539a429a37d14be904"
+}
+{
+  "name": "che-theia-7.20.1",
+  "image": "quay.io/eclipse/che-theia@sha256:a56428ee28ceefca2a1cba51093f6015e0eee82659fec786f2e1fdc939fdb856"
+}
+...
```
Comment (rkratky) [inline modules/administration-guide/examples/che-prepulled-images-table.adoc]:
```suggestion
#[...]
```

File: modules/administration-guide/examples/che-prepulled-images-table.adoc [hunk 10]
```diff
@@ -22,4 +22,53 @@
...
+----
++
+Following images from the list cannot be pre-pulled, therefore need to be pulled manually with `skopeo copy --all __<...>__`:
```
Comment (rkratky) [inline modules/administration-guide/examples/che-prepulled-images-table.adoc]:
```suggestion
The following images from the list cannot be pre-pulled and need to be pulled manually using:

[subs="+quotes"]
----
$ skopeo copy --all _<image>_
----
```

File: modules/administration-guide/examples/che-prepulled-images-table.adoc [hunk 11]
```diff
@@ -22,4 +22,53 @@
...
++
+* machine-exec image
```
Comment (rkratky) [inline modules/administration-guide/examples/che-prepulled-images-table.adoc:78]: List all these images by name (without versions) to make it clear, which ones are they.

File: modules/administration-guide/examples/che-prepulled-images-table.adoc [hunk 12]
```diff
@@ -22,4 +22,53 @@
...
++
+----
+# dnf install -y jq python-pip
+----
++
+----
+# pip install yq
+----
+
+. Use the following scriptlet:
```
Comment (rkratky) [inline modules/administration-guide/examples/che-prepulled-images-table.adoc:45]:
```suggestion
.Procedure

. Use the following scriptlet:
```

**Review by rkratky — CHANGES_REQUESTED** [review]

File: antora-playbook.yml [hunk 13]
```diff
@@ -96,12 +96,13 @@ asciidoc:
...
     prod-prev-ver-major: 6
     prod-prev-ver: 7.18
+    che-ver: 7.21.1
     prod-short: Che
     prod-upstream: Eclipse{nbsp}Che
     prod-url-secure: \https://che-host
     prod-url: \https://che-host:che-port
     prod-ver-major: 7
-    prod-ver: 7.19
+    prod-ver: 7.21
```
Comment (nickboldt) [inline antora-playbook.yml:105]: should bump to latest 7.24

**Comment by themr0c** [comment]
Superceded by https://github.com/eclipse/che-docs/pull/1782

**Pull request closed without merging**